lib/msal-core/src/cache/AuthCache.ts

```diff
@@ -153,11 +154,16 @@ export class AuthCache extends BrowserStorage {// Singleton
      */
     getAllAccessTokens(clientId: string, homeAccountIdentifier: string): Array<AccessTokenCacheItem> {
         const results = Object.keys(window[this.cacheLocation]).reduce((tokens, key) => {
-            if ( window[this.cacheLocation].hasOwnProperty(key) && key.match(clientId) && key.match(homeAccountIdentifier)) {
+            if ( window[this.cacheLocation].hasOwnProperty(key) && key.match(clientId) && key.match(homeAccountIdentifier) && key.match(Constants.scopes)) {
```

Comment:
I realize this was there before, but `window[this.cacheLocation].hasOwnProperty(key)` isn't needed as it will always be true given the nature of the `reduce`.

Reply:
+1

Reply:
+1 done